cli/go.mod

```diff
@@ -1,8 +1,11 @@
 module github.com/wordpress-mobile/gbm-cli
 
-go 1.20
+go 1.21
```

Comment:
I was prompted to add these changes to `go.mod` after running `go run main.go release prepare v1.0.0`, which asked me to run `go mod tidy`. I think it may be because I was using Go `v1.21`, when the go.mod file was set on Go `v1.20`. These dependency updates seem fine, but let me know if there's something to update here -- or perhaps there's a way to enforce a common Go version for the project, similar to `nvm` checks? Overall I think it's a minor point, but here's the command output I was following that prompted this change:

```
release-toolkit-gutenberg-mobile/cli on  cli/add-prepare-cmd [$] via 🐹 v1.21.2 
➜ GBM_WORDPRESS_ORG=derekblank go run main.go release prepare v1.0.0
go: updates to go.mod needed; to update it:
        go mod tidy

release-toolkit-gutenberg-mobile/cli on  cli/add-prepare-cmd [$] via 🐹 v1.21.2 
➜ go mod tidy
go: downloading github.com/stretchr/testify v1.7.0
go: downloading gopkg.in/h2non/gock.v1 v1.1.2
go: downloading github.com/inconshreveable/mousetrap v1.1.0
go: downloading golang.org/x/image v0.6.0
go: downloading golang.org/x/mobile v0.0.0-20230301163155-e0f57694e12c
go: downloading gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/kr/pretty v0.1.0
go: downloading golang.org/x/exp v0.0.0-20190731235908-ec7cb31e5a56
go: downloading github.com/h2non/parth v0.0.0-20190131123155-b4df798d6542
go: finding module for package github.com/kr/text
go: downloading github.com/kr/text v0.2.0
go: found github.com/kr/text in github.com/kr/text v0.2.0
```

Reply:
Hmm not sure what to do here. I did notice that the `go.mod` file no longer works with go v1.20. It was complaining about the `toolchain` directive.

I updated my go version to 1.21.3 and it seems to be working again. Doing this just to unblock the work but we should try to figure out a better way to manage the go version.

Reply:
here are the docs on `toolchain` for future reference https://go.dev/doc/toolchain

Reply:
So this kinda messed up my vscode setup.  I was unable to get vscode to recognize go v1.21.3, even after removing v1.20 from my machine. 
I had to remove `GOROOT` from my zsh config per the advice in this SO post https://stackoverflow.com/a/76321397/1373043

Reply:
We can revert this commit if we need to, and stick with Go `v1.20`. It wouldn't let me _not_ run `go mod tidy` on `v1.21`, however.